internal/server/server.go

```diff
@@ -338,13 +338,24 @@ func (s *Server) handleLogin() http.HandlerFunc {
 			return
 		}
 		session.Values["nonce"] = nonce
+		// generate a PKCE code verifier
+		codeVerifier := oauth2.GenerateVerifier()
+		codeChallenge := oauth2.S256ChallengeFromVerifier(codeVerifier)
+
+		// store code_verifier for token exchange
+		session.Values["code_verifier"] = codeVerifier
```

Comment:
The code_verifier replaces the need for the "state" management in the root and login handlers.

Reply:
Are you referring to the state or the session here? Both are passed around the login and callback handlers